src/Shared/RoslynUtils/WellKnownTypes.cs

```diff
@@ -51,6 +52,7 @@ private WellKnownTypes(Compilation compilation)
     {
         _lazyWellKnownTypes = new INamedTypeSymbol?[WellKnownTypeData.WellKnownTypeNames.Length];
         _compilation = compilation;
+        _missingTypeSymbol = compilation.GetTypeByMetadataName(typeof(MissingType).FullName!)!;
```

Comment:
I wonder if we should consider reusing SpecialType.None as the fallback in this case as Roslyn does in their WellKnownType implementation ([ref](https://github.com/dotnet/roslyn/blob/322037893f72b11ce30cb14bb1727798f0a7eb4a/src/Compilers/Core/Portable/WellKnownTypes.cs#L19)). That let's us reuse an existing concept instead of defining out own marker type.